hazelcast/src/test/java/com/hazelcast/spi/impl/operationservice/impl/InvocationNestedLocalTest.java

```diff
@@ -8,115 +8,120 @@
 import com.hazelcast.test.HazelcastParallelClassRunner;
 import com.hazelcast.test.annotation.ParallelTest;
 import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Rule;
 import org.junit.Test;
 import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
 import org.junit.runner.RunWith;
 
 import java.util.concurrent.ExecutionException;
-import java.util.concurrent.Future;
 
 import static org.junit.Assert.assertEquals;
-import static org.junit.Assert.fail;
 
 @RunWith(HazelcastParallelClassRunner.class)
 @Category({QuickTest.class, ParallelTest.class})
 public class InvocationNestedLocalTest extends InvocationNestedTest {
 
-    private final String response = "someresponse";
+    private static final String RESPONSE = "someresponse";
+
+    @Rule
+    public ExpectedException expected = ExpectedException.none();
 
     @Test
-    public void whenPartition_callsGeneric() {
-        HazelcastInstance hz = createHazelcastInstance();
-        OperationService operationService = getOperationService(hz);
+    public void invokeOnPartition_outerGeneric_innerGeneric_forbidden() {
+        HazelcastInstance local = createHazelcastInstance();
+        OperationService operationService = getOperationService(local);
+
+        InnerOperation innerOperation = new InnerOperation(RESPONSE, GENERIC_OPERATION);
+        OuterOperation outerOperation = new OuterOperation(innerOperation, GENERIC_OPERATION);
+
+        expected.expect(Exception.class);
```

Comment:
just friendly advice :)
in general, in hazelcast we use (expected = SomeExceptionClassName) pattern.
You can see an example here: https://github.com/hazelcast/hazelcast/blob/08857da91e497365437700243db026fda4c5c31a/hazelcast/src/test/java/com/hazelcast/config/QueryCacheConfigTest.java#L15